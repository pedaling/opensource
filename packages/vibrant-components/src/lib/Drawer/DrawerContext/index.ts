export { DrawerProvider, useDrawer } from './DrawerContext';
