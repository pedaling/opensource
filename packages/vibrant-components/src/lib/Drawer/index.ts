export { Drawer } from './Drawer';
export * as DrawerTranslation from './translation';
