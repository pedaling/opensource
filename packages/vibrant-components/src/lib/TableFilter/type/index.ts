export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
