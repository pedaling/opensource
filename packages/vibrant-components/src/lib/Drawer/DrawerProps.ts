import type { ReactElement } from 'react';
import { withVariation } from '@vibrant-ui/core';

export type DrawerProps = {
  testId?: string;
  children: ReactElement[];
  type?: 'modal' | 'overlay' | 'standard';
  placement: 'bottom' | 'left' | 'right' | 'top';
  open?: boolean;
  onOpen?: () => void;
  onClose?: () => void;
};

export const withDrawerVariation = withVariation<DrawerProps>('Drawer')();
