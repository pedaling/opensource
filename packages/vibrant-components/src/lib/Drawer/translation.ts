import type { Translation } from '@vibrant-ui/core';

export const ko: Translation['drawer'] = {
  drawerHeader: {
    ariaLabel: '닫기',
  },
};

export const en: Translation['drawer'] = {
  drawerHeader: {
    ariaLabel: 'Close',
  },
};

export const ja: Translation['drawer'] = {
  drawerHeader: {
    ariaLabel: '閉じる',
  },
};
