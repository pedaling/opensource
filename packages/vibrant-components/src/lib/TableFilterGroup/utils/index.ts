import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
