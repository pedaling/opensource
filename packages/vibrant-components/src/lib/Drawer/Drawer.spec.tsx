import { waitFor } from '@testing-library/react';
import { Box } from '@vibrant-ui/core';
import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Drawer } from './Drawer';

describe('<Drawer />', () => {
  const { render } = createReactRenderer();

  describe('when Drawer is rendering each placement', () => {
    let placementRightRenderer: ReactRenderer;
    let placementLeftRenderer: ReactRenderer;
    let placementTopRenderer: ReactRenderer;
    let placementBottomRenderer: ReactRenderer;

    beforeEach(async () => {
      placementRightRenderer = render(
        <Drawer type="overlay" placement="right" open={true}>
          <Box />
          <Drawer.Panel defaultSize={320}>
            <Box />
          </Drawer.Panel>
        </Drawer>
      );

      placementLeftRenderer = render(
        <Drawer type="overlay" placement="left" open={true}>
          <Box />
          <Drawer.Panel defaultSize={320}>
            <Box />
          </Drawer.Panel>
        </Drawer>
      );

      placementTopRenderer = render(
        <Drawer type="overlay" placement="top" open={true}>
          <Box />
          <Drawer.Panel defaultSize={320}>
            <Box />
          </Drawer.Panel>
        </Drawer>
      );

      placementBottomRenderer = render(
        <Drawer type="overlay" placement="bottom" open={true}>
          <Box />
          <Drawer.Panel defaultSize={320}>
            <Box />
          </Drawer.Panel>
        </Drawer>
      );
    });

    it('match snapshot', () => {
      expect(placementRightRenderer.container).toMatchSnapshot();

      expect(placementLeftRenderer.container).toMatchSnapshot();

      expect(placementTopRenderer.container).toMatchSnapshot();

      expect(placementBottomRenderer.container).toMatchSnapshot();
    });
  });

  describe('when Drawer type is standard', () => {
    let renderer: ReactRenderer;
    let container: HTMLElement;
    let computedStyle: CSSStyleDeclaration;

    beforeEach(async () => {
      renderer = render(
        <Box width={880}>
          <Drawer type="standard" placement="right" open={true}>
            <Box />
            <Drawer.Panel defaultSize={400}>
              <Box />
            </Drawer.Panel>
          </Drawer>
        </Box>
      );

      container = renderer.getByTestId('drawer-content-container');

      computedStyle = getComputedStyle(container);
    });

    it('content size is shrink', async () => {
      await waitFor(() => expect(computedStyle.width).toEqual('480px'));
    });
  });

  describe('when Drawer type is modal', () => {
    let renderer: ReactRenderer;

    beforeEach(async () => {
      renderer = render(
        <Drawer type="modal" placement="right" open={true}>
          <Box />
          <Drawer.Panel defaultSize={320}>
            <Box />
          </Drawer.Panel>
        </Drawer>
      );
    });

    it('dim element is activate', () => {
      expect(renderer.getByTestId('dim-background')).toHaveStyleRule('background-color', '#000000b2');
    });
  });
});
