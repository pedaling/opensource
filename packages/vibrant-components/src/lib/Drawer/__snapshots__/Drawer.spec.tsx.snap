// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> when Drawer is rendering each placement match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  overflow: auto;
  width: 100%;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: absolute;
  height: 100%;
  background-color: #ffffff;
  top: 0;
  right: 0;
  z-index: 3;
  box-shadow: 0px 0px 1px 0px #00000014,0px 10px 25px 4px #0000000f;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  width: 320px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 100%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-width: 0;
  border-color: #00000026;
  border-style: solid;
  width: 0;
  border-right-width: 1px;
  height: 100%;
  margin-left: 0;
  margin-right: 0;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  overflow: hidden;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer"
    tabindex="0"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-label="drawer-content-container"
        class="emotion-2"
      >
        <div
          class="emotion-3"
        />
      </div>
      <div
        class="emotion-4"
        style="transform: translateX(0px); transition-property: transform; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
      >
        <div
          class="emotion-5"
          data-testid="drawer-panel"
        >
          <div
            class="emotion-6"
            kind="default"
          />
          <div
            class="emotion-7"
          >
            <div
              class="emotion-8"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-3"
                />
              </div>
            </div>
            <div
              class="emotion-11"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Drawer /> when Drawer is rendering each placement match snapshot 2`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  overflow: auto;
  width: 100%;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: absolute;
  height: 100%;
  background-color: #ffffff;
  top: 0;
  left: 0;
  z-index: 3;
  box-shadow: 0px 0px 1px 0px #00000014,0px 10px 25px 4px #0000000f;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  width: 320px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 100%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-width: 0;
  border-color: #00000026;
  border-style: solid;
  width: 0;
  border-right-width: 1px;
  height: 100%;
  margin-left: 0;
  margin-right: 0;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  overflow: hidden;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer"
    tabindex="0"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-label="drawer-content-container"
        class="emotion-2"
      >
        <div
          class="emotion-3"
        />
      </div>
      <div
        class="emotion-4"
        style="transform: translateX(0px); transition-property: transform; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
      >
        <div
          class="emotion-5"
          data-testid="drawer-panel"
        >
          <div
            class="emotion-6"
            kind="default"
          />
          <div
            class="emotion-7"
          >
            <div
              class="emotion-8"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-3"
                />
              </div>
            </div>
            <div
              class="emotion-11"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Drawer /> when Drawer is rendering each placement match snapshot 3`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  overflow: auto;
  width: 100%;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: absolute;
  width: 100%;
  background-color: #ffffff;
  top: 0;
  left: 0;
  z-index: 3;
  box-shadow: 0px 0px 1px 0px #00000014,0px 10px 25px 4px #0000000f;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  height: 320px;
  width: 100%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-width: 0;
  border-color: #00000026;
  border-style: solid;
  width: 100%;
  height: 0;
  border-top-width: 1px;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  overflow: hidden;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer"
    tabindex="0"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-label="drawer-content-container"
        class="emotion-2"
      >
        <div
          class="emotion-3"
        />
      </div>
      <div
        class="emotion-4"
        style="transform: translateY(0px); transition-property: transform; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
      >
        <div
          class="emotion-5"
          data-testid="drawer-panel"
        >
          <div
            class="emotion-6"
            kind="default"
          />
          <div
            class="emotion-7"
          >
            <div
              class="emotion-8"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-3"
                />
              </div>
            </div>
            <div
              class="emotion-11"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Drawer /> when Drawer is rendering each placement match snapshot 4`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  overflow: auto;
  width: 100%;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: absolute;
  width: 100%;
  background-color: #ffffff;
  bottom: 0;
  left: 0;
  z-index: 3;
  box-shadow: 0px 0px 1px 0px #00000014,0px 10px 25px 4px #0000000f;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  height: 320px;
  width: 100%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-width: 0;
  border-color: #00000026;
  border-style: solid;
  width: 100%;
  height: 0;
  border-top-width: 1px;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  overflow: hidden;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer"
    tabindex="0"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-label="drawer-content-container"
        class="emotion-2"
      >
        <div
          class="emotion-3"
        />
      </div>
      <div
        class="emotion-4"
        style="transform: translateY(0px); transition-property: transform; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
      >
        <div
          class="emotion-5"
          data-testid="drawer-panel"
        >
          <div
            class="emotion-6"
            kind="default"
          />
          <div
            class="emotion-7"
          >
            <div
              class="emotion-8"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-3"
                />
              </div>
            </div>
            <div
              class="emotion-11"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
